src/components/dialog/DialogHeaderSearch.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { ReactElement, useState } from 'react'
+import { ClassNameProps, UseSearchQuery } from '../types'
+import { HeaderNoSearch } from './headerSearch/HeaderNoSearch'
+import { HeaderSearch } from './headerSearch/HeaderSearch'
+
+type Props = {
+  /**
+   * Defines which action should be executed on dismissing.
+   **/
+  onDismiss: React.MouseEventHandler
+  /**
+   * Defines the title of the header.
+   **/
+  title: string
+  /**
+   * Define the accessibility label for the search icon.
+   **/
+  iconLabel: string
+} & UseSearchQuery &
+  ClassNameProps
+
+export function DialogHeaderSearch({
```

Comment:
```suggestion
export function DialogHeaderWithSearch({
```

Same here. What do you think?

Reply:
Let's discuss this above with the `DialogHeaderClose` component :)